backend/tournesol/views/preview.py

```diff
@@ -260,14 +313,339 @@ def get(self, request, uid):
             return self.default_preview()
 
         response = HttpResponse(content_type="image/png")
-        preview_image = get_preview_frame(entity, get_preview_font_config())
+
+        upscale_ratio = 2
+        fnt_config = get_preview_font_config(upscale_ratio=upscale_ratio)
+        preview_image = get_preview_frame(
+            entity, fnt_config, upscale_ratio=upscale_ratio
+        )
 
         try:
-            youtube_thumbnail = self.get_yt_thumbnail(entity)
+            youtube_thumbnail = self.get_yt_thumbnail(entity, quality="maxres")
         except ConnectionError:
             return self.default_preview()
 
-        preview_image.paste(youtube_thumbnail, box=(120, 0))
-        self._draw_logo(preview_image, entity)
+        youtube_thumbnail = youtube_thumbnail.resize(
+            numpy.multiply((320, 180), upscale_ratio)
+        )
+        preview_image.paste(
+            youtube_thumbnail, box=tuple(numpy.multiply((120, 0), upscale_ratio))
+        )
+
+        self._draw_logo(preview_image, entity, upscale_ratio=upscale_ratio)
+
         preview_image.save(response, "png")
         return response
+
+
+class ComparisonPreviewGenerator:
+    def __init__(self):
+        # An upscale ratio > 1 generates a bigger image while keeping all the relative sizes
+        # and positions identical (like a zoom). Only integer values are allowed for now.
+        self.upscale_ratio = 2
+
+    def render(self, entity_a, entity_b, thumbnail_a, thumbnail_b):
+        # pylint: disable=too-many-locals
+
+        base_size = (440, 240)
+
+        # All the sizes and positions in this method are relative to the base size, not the
+        # upscaled size. The other methods are responsible for calculating the final (upscaled)
+        # sizes and positions.
+
+        base_margin = 6
+        horizontal_separator_height = base_margin
+        margin_above_entity_descriptions = base_margin
+        entity_descriptions_horizontal_margin = base_margin
+        logo_size = 32
+        margin_above_slider = base_margin * 3
+        margin_below_slider = base_margin
+        vs_band_width = 34
+        vs_tilt_in_pixels = 10
+        text_below_slider = "Which one should be largely recommended?"
+
+        final = self.generate_image(base_size, color=COLOR_WHITE_BACKGROUND)
+
+        thumbnail_size = self.draw_thumbnail(
```

Comment:
Should we avoid an assignation here, if the name `thumbnail_size` is re-assigned another value just after?

Reply:
Changed in 23e421493818307919edeedf40e3b62828de6714 too.